### PR TITLE
Add C++ (Modern/UWP,MFC, etc.) support for SemVer in GitVersion

### PR DESCRIPTION
**Pull request by stunney**
A few simple steps to get this working
1) Create a file called "GitVersionTaskAssemblyInfo.h" right beside your .vcxproj
2) Fill it with this:
#pragma once

#define VERSION_STR(x)			#x
#define VERSION_XSTR(x)			VERSION_STR(x)
#define VERSION_BLANK ""
#define VERSION_FULL 0,0,0,0
#define VERSION_STRING VERSION_XSTR(0.0.0.0) VERSION_BLANK "\0"

3) Create a version resource (filename does not matter)
4) Modify the .rc code:
++ Add "#include "GitVersionAssemblyInfo.g.h" somewhere near the top
Modify the versioninfo lines to be (lines 53,54 in a typical, default Version resouce file):
FILEVERSION VERSION_FULL
PRODUCTVERSION VERSION_FULL

and

Modify the block for the PE description as follows for the FileVersion and ProductVersion lines

VALUE "FileVersion", VERSION_FULL
VALUE "ProductVersion", VERSION_STRING

5) Add your GitVersion.yml to your heart's delight.

6) Drink coffee (or beverage of choice).

### CODE REVIEW

File: src/GitVersionCore/GitVersionCore.csproj [hunk 1]
```diff
@@ -47,7 +47,9 @@
       <HintPath>..\packages\LibGit2Sharp.0.23.0-pre20160922233542\lib\net40\LibGit2Sharp.dll</HintPath>
       <Private>True</Private>
     </Reference>
-    <Reference Include="System" />
+    <Reference Include="System">
+      <HintPath>..\..\..\..\..\..\..\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.0\System.dll</HintPath>
```
Comment (denis-ivanov) [inline src/GitVersionCore/GitVersionCore.csproj]: I think you don't need to reference assembly by explicit path.

Comment (stunney) [reply src/GitVersionCore/GitVersionCore.csproj]: Bah!  I blame visual studio for this abomination!  Thanks.  I will just revert this to the original line.

**Comment by stunney** [comment]
Bah!  Time to re-base and resubmit...  Apologies guys!

**Comment by stunney** [comment]
@denis-ivanov I have submitted a new PR.  Can you please review?

I did not realize that your VS projects were on steroids and I didn't actually need to modify them.  I'm not used to that from VS projects.  This pleases me.

**Pull request closed without merging**